modules/images/webp-uploads/load.php

```diff
@@ -5,40 +5,294 @@
  * Experimental: No
  *
  * @package performance-lab
- * @since 1.0.0
+ * @since   1.0.0
  */
 
 /**
- * Filter the image editor default output format mapping.
+ * Hook called by `wp_generate_attachment_metadata` to create the `sources` property for every image
+ * size, the sources' property would create a new image size with all the mime types specified in
+ * `webp_uploads_valid_image_mime_types`. If the original image is one of the mimes from
+ * `webp_uploads_valid_image_mime_types` the image is just added to the `sources` property and  not
+ * created again. If the uploaded attachment is not a valid image this function does not alter the
+ * metadata of the attachment, on the other hand a `sources` property is added.
  *
- * For uploaded JPEG images, map the default output format to WebP.
+ * @since n.e.x.t
  *
- * @since 1.0.0
+ * @see   wp_generate_attachment_metadata
+ * @see   webp_uploads_valid_image_mime_types
  *
- * @param string $output_format The image editor default output format mapping.
- * @param string $filename      Path to the image.
- * @param string $mime_type     The source image mime type.
- * @return string The new output format mapping.
+ * @param array $metadata      An array with the metadata from this attachment.
+ * @param int   $attachment_id The ID of the attachment where the hook was dispatched.
+ *
+ * @return array An array with the updated structure for the metadata before is stored in the database.
  */
-function webp_uploads_filter_image_editor_output_format( $output_format, $filename, $mime_type ) {
-	// Only enable if the server supports WebP.
-	if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
-		return $output_format;
+function webp_uploads_create_sources_property( array $metadata, $attachment_id ) {
+	// This should take place only on the JPEG image.
+	$valid_mime_transforms = webp_uploads_get_supported_image_mime_transforms();
+
+	// Not a supported mime type to create the sources property.
+	if ( ! array_key_exists( get_post_mime_type( $attachment_id ), $valid_mime_transforms ) ) {
```

Comment:
Nit-pick, but `isset` is a tiny bit faster than `array_key_exists`, and the latter is not needed here.
```suggestion
	$mime_type = get_post_mime_type( $attachment_id );
	if ( ! isset( $valid_mime_transforms[ $mime_type ] ) ) {
```

Reply:
Good point.